core/src/main/java/org/elasticsearch/action/support/ActiveShardCount.java

```diff
@@ -160,12 +160,13 @@ public boolean enoughShardsActive(final ClusterState clusterState, final String
      * to meet the required shard count represented by this instance.
      */
     public boolean enoughShardsActive(final IndexShardRoutingTable shardRoutingTable) {
+        final int activeShardCount = shardRoutingTable.activeShards().size();
         if (this == ActiveShardCount.ALL) {
-            return shardRoutingTable.allShardsStarted();
+            return activeShardCount == shardRoutingTable.replicaShards().size() + 1;
```

Comment:
Perhaps add a comment to explain why the +1? (because of the primary I assume, but it's still nice to be explicit for newcomers to the codebase)